internal/provider/resources/install/aws/aws_iam_write.go

```diff
@@ -0,0 +1,400 @@
+														),
+													),
+												},
+											},
+											"parent": schema.StringAttribute{
+												Optional:            true,
+												MarkdownDescription: `The account ID of the federation app's parent AWS account`,
+												Validators: []validator.String{
+													stringvalidator.RegexMatches(AwsAccountIdRegex, "AWS account IDs should consist of 12 numeric digits"),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						Validators: []validator.Object{
+							objectvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("identity")),
+							objectvalidator.ConflictsWith(path.MatchRelative().AtParent().AtName("parent")),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *AwsIamWrite) getId(data any) *string {
+	model, ok := data.(*AwsIamWriteModel)
+	if !ok {
+		return nil
+	}
+	return &model.Id
+}
+
+func (r *AwsIamWrite) getItemJson(json any) any {
+	inner, ok := json.(*AwsIamWriteApi)
+	if !ok {
+		return nil
+	}
+	return inner.Item
+}
+
+func (r *AwsIamWrite) fromJson(id string, json any) any {
+	data := AwsIamWriteModel{}
+	jsonv, ok := json.(*AwsIamWriteJson)
+	if !ok {
+		return nil
+	}
+
+	data.Id = id
+	data.Label = types.StringNull()
+	if jsonv.Label != nil {
+		data.Label = types.StringValue(*jsonv.Label)
+	}
+
+	data.State = types.StringValue(jsonv.State)
+	data.Login = jsonv.Login
+
+	return &data
+}
+
+func (r *AwsIamWrite) toJson(data any) any {
+	json := AwsIamWriteJson{}
+
+	datav, ok := data.(*AwsIamWriteModel)
+	if !ok {
+		return nil
+	}
+
+	if !datav.Label.IsNull() && !datav.Label.IsUnknown() {
+		label := datav.Label.ValueString()
+		json.Label = &label
+	}
+
+	// can omit state here as it's filled by the backend
+	json.Login = datav.Login
+
+	return &json
+}
+
+func (r *AwsIamWrite) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	providerData := internal.Configure(&req, resp)
+	r.installer = &installresources.Install{
+		Component:    IamWrite,
+		Integration:  Aws,
+		ProviderData: providerData,
+		GetId:        r.getId,
+		GetItemJson:  r.getItemJson,
+		FromJson:     r.fromJson,
+		ToJson:       r.toJson,
+	}
+}
+
+func (r *AwsIamWrite) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var json AwsIamWriteApi
+	var data AwsIamWriteModel
+	r.installer.Upsert(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &json, &data)
+}
+
+func (r *AwsIamWrite) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var json AwsIamWriteApi
+	var data AwsIamWriteModel
+	r.installer.Read(ctx, &resp.Diagnostics, &resp.State, &json, &data)
+}
+
+func (r *AwsIamWrite) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var json AwsIamWriteApi
+	var data AwsIamWriteModel
+	r.installer.Upsert(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &json, &data)
```

Comment:
If we're not using these values we can inline them:
```
r.installer.Upsert(ctx, &resp.Diagnostics, &req.Plan, &resp.State, &AwsIamWriteApi{}, &AwsIamWriteModel{})
```

Reply:
TIL